scene/animation/animation_tree.h

```diff
@@ -96,10 +97,14 @@ class AnimationNode : public Resource {
 	HashMap<NodePath, bool> filter;
 	bool filter_enabled = false;
 
+	StringName node_name;
+	AnimationRootNode *tree_root = nullptr;
```

Comment:
Raw backreferences are scary. This will be the source of too many bugs. We mustn't do this in this way.

I need better justification why this is needed, and we need to guarnatee that 
all codepaths in which a node is freed or removed properly sanitize these raw backreferences, including in cases where nodes are shared between multiple trees (not the right way to use animation tree, but not illegal either: we don't want Godot to crash if this happens)

Reply:
The "blackboard" parameters are stored in the `tree_root` node, so each `AnimationNode` needs to know who its `tree_root` is in order to access the global parameters that this PR introduces.